docs/csharp/tutorials/working-with-linq.md

```diff
@@ -174,49 +205,58 @@ Let's see how many shuffles it takes to set the deck back to its original order.
 
 You can see this in action when you use it to determine when the deck is back in its original order. Put the shuffle code inside a loop, and stop when the sequence is back in its original order by applying the `SequenceEquals()` method. You can see it would always be the final method in any query, because it returns a single value instead of a sequence:
 
 ```csharp
-var times = 0;
-var shuffle = startingDeck;
-
-do
+// Program.cs
+static void Main(string[] args)
 {
-    shuffle = shuffle.Take(26).InterleaveSequenceWith(shuffle.Skip(26));
+    // Query for building the deck
 
-    foreach (var c in shuffle)
+    // Shuffling using InterleaveSequenceWith<T>();
+
+    var times = 0;
+    // We can re-use the shuffle variable from earlier, or you can make a new one
+    shuffle = startingDeck;
+    do
     {
-        Console.WriteLine(c);
-    }
+        shuffle = shuffle.Take(26).InterleaveSequenceWith(shuffle.Skip(26));
 
-    Console.WriteLine();
-    times++;
-} while (!startingDeck.SequenceEquals(shuffle));
+        foreach (var card in shuffle)
+        {
+            Console.WriteLine(card);
+        }
+        Console.WriteLine();
+        times++;
+
+    } while (!startingDeck.SequenceEquals(shuffle));
 
-Console.WriteLine(times);
+    Console.WriteLine(times);
+}
 ```
 
-Run the sample, and see how the deck rearranges on each shuffle, until it returns to its original configuration after 8 iterations.
+Run the code we've got so far. Notice how the deck rearranges on each shuffle; after 8 shuffles (iterations of the do-while loop), the deck returns to its original configuration it was in when we first created it from our starting LINQ query.
 
 ## Optimizations
 
-The sample you've built so far executes an *out shuffle*, where the top and bottom cards stay the same on each run. Let's make one change, and run an *in shuffle*, where all 52 cards change position. For an in shuffle, you interleave the deck so that the first card in the bottom half becomes the first card in the deck. That means the last card in the top half becomes the bottom card. That's just a one line change. Update the call to shuffle to change the order of the top and bottom halves of the deck:
+The sample you've built so far executes an *outer shuffle*, where the top and bottom cards stay the same on each run. Let's make one change: we'll use an *inner shuffle* instead, where all 52 cards change position. For an inner shuffle, you interleave the deck so that the first card in the bottom half becomes the first card in the deck. That means the last card in the top half becomes the bottom card. This is a simple change to a singular line of code. Update our current shuffle query by switching the positions of <xref:System.Linq.Enumerable.Take%2A> and <xref:System.Linq.Enumerable.Skip%2A>. This will change the order of the top and bottom halves of the deck:
```

Comment:
The terms "in shuffle" and "out shuffle" were correct. (I know, it sounds weird to me too).